watchdog_reader.go

```diff
@@ -23,9 +25,14 @@ func newWatchdogReader(reader io.Reader, timeout time.Duration, timer *time.Time
 
 // Read reads up to len(p) bytes into p
 func (t *watchdogReader) Read(p []byte) (n int, err error) {
-	// FIXME limit the amount of data read in one chunk so as to not exceed the timeout?
+	//never read more bytes than watchdogChunkSize
+	readTarget := p
+	if len(p) > watchdogChunkSize {
+		readTarget = p[0:watchdogChunkSize]
+	}
+
```

Comment:
I'm not sure if even calling `t.reader.Read()` in a loop would help here. In any case, having a smaller buffer would cause more read() syscalls to be issued to the underlying socket, and that's where I'd guess the biggest performance penalty would come from.

Maybe one solution would be having the `watchdogChunkSize` configurable in the `Connection` struct and some docs to explain its relationship to the `Timeout` attribute? Not sure if this is a good idea though as the Connection struct already seems a bit crowded.